### DISCUSSION TITLE
🐛 BUG: `wrangler pages dev` `--live-reload` does not watch static assets

### DISCUSSION THREAD

**Discussion by malobre** [cloudflare/workers-sdk#13502] [Feature Requests]
### Which Cloudflare product(s) does this pertain to?

Wrangler core, Miniflare

### What version(s) of the tool(s) are you using?

wrangler@3.41.0, miniflare@3.20240320.1

### What version of Node are you using?

21.7.0

### What operating system and version are you using?

Linux 6.8.1 (arch linux)

### Observed behavior
Editing a file served by the pages dev server does not trigger a page reload.

### Expected behavior
Page reload on file update.

### Steps to reproduce
- create a file (in my case a simple `index.html`)
- start pages dev server `wrangler pages dev --live-reload [directory]`
- edit the file

### misc
There are comments on the original PR adding the live reload feature mentioning that users are unable to get it to work: #220 

Potentially related to #4124 ?

### Please provide any relevant error logs

No errors in browser console or wrangler logs

**Comment by malobre** [comment]
After a little investigation wrangler seems to only watch files in the `functions` dir,
e.g: with `wrangler pages dev --live-reload dist`:
- `touch dist/my-page.html` doesn't reload `/my-page`
- `touch functions/my-page.ts` *does* reload `/my-page`

**Comment by treeder** [comment]
That would explain this one too: https://github.com/cloudflare/workers-sdk/issues/4824

**Comment by malobre** [comment]
Yes, this is surely related, if not a duplicate.

**Comment by CarmenPopoviciu** [comment]
Hi @malobre, thank you for reporting this issue. 

You are indeed correct in saying that `pages dev` does not watch for static assets changes. It currently watches for:
- `/functions` directory
- `_worker.js` file/directory
- `_routes.json`

Triaging this issue and will discuss it as part of the `pages dev` improvements [milestone](https://github.com/cloudflare/workers-sdk/milestone/10)

**Comment by Hades32** [comment]
this, combined with `--live-reload` reloading before some build processes finish, makes that feature barely usable

**Comment by serban-mihai** [comment]
Are there any updates on this?
It's a major blocker for smooth development in local environments.

**Comment by zugdev** [comment]
what I've done is run another server&builder, using wrangler only for cloudflare related functions, in my case I use esbuild

**Comment by serban-mihai** [comment]
> what I've done is run another server&builder, using wrangler only for cloudflare related functions, in my case I use esbuild

I know there are endless combinations of services that could be applied to Pages in such a scenario, but if you could provide an example that would be much appreciated.

**Comment by Brennvo** [comment]
Anecdotally, this was particularly confusing to me because the documentation for `--live-reload` mentions HTML pages:

> `--live-reload`
> Auto reload HTML pages when change is detected

However, based off @CarmenPopoviciu's comment:

> You are indeed correct in saying that pages dev does not watch for static assets changes. It currently watches for:
> 
> * `/functions` directory
> * `_worker.js` file/directory
> * `_routes.json`

This suggests HTML only reloads when these type of files change. Maybe this could also be cleared up in docs?
